drivers/digital-io/max22196/max22196.c

```diff
@@ -198,6 +198,9 @@ int max22196_set_mode(struct max22196_desc *desc, uint32_t ch,
 	if (ch > MAX22196_CHANNELS - 1)
 		return -EINVAL;
 
+	if (desc->chip_id == ID_MAX22194 && ch > MAX22194_CHANNELS - 1)
```

Comment:
all in all the code would look much nicer if you would have in desc the maximum channels numbers which you add at init based on device_id... then you simply check ch against that number...